Project/Assets/ML-Agents/Examples/Match3/Scripts/Match3Board.cs

```diff
@@ -23,16 +35,25 @@ public class Match3Board : AbstractBoard
         /// </summary>
         public int RandomSeed;
 
-        (int, int)[,] m_Cells;
+        (int CellType, int SpecialType)[,] m_Cells;
         bool[,] m_Matched;
 
+        private BoardSize m_CurrentBoardSize;
+
         System.Random m_Random;
 
         void Awake()
         {
-            m_Cells = new (int, int)[Columns, Rows];
-            m_Matched = new bool[Columns, Rows];
+            m_Cells = new (int, int)[MaxColumns, MaxRows];
+            m_Matched = new bool[MaxColumns, MaxRows];
 
+            m_CurrentBoardSize = new BoardSize
```

Comment:
sounds a bit weird that BoardSize has attributes like NumCellTypes and NumSpecialTypes. BoardConfig?

Reply:
Also, maybe it's convoluted logic, but the NumCellTypes and NumSpecialTypes define the "depth" of the board observations.

Reply:
yea BoardSpec sounds good, so that you can say MaxBoardSpec for max size.

But you convinced me with the "depth" logic. I'm okay with both, leaning towards BoardSpec.

Reply:
I think I convinced myself too :)  I'm leaning towards BoardSize; I think it makes a more sense to talk about "maximum" for board sizes, but not for specs.